core/src/main/java/org/aerogear/mobile/core/MobileCore.java

```diff
@@ -142,7 +136,7 @@ public void destroy() {
         }
 
         try {
-            ServiceModule serviceModule = serviceClass.newInstance();
+            final ServiceModule serviceModule = serviceClass.newInstance();
 
             if (serviceConfiguration == null) {
                 serviceConfiguration = getServiceConfiguration(serviceModule.type());
```

Comment:
Changing parameters value is a bad practice

Reply:
yeah - for a quick fix, I am doing a local var